### PR TITLE
diff stats: fix segfaults with new files

### PR DESCRIPTION
**Pull request by ddevault**

### CODE REVIEW

**Comment by ddevault** [comment]
We're tracking this releated bug on lists.sr.ht:

https://todo.sr.ht/~sircmpwn/lists.sr.ht/154

**Comment by ddevault** [comment]
bump

**Comment by ethomson** [comment]
Ah sorry I didn't submit my comment

File: src/diff_stats.c [hunk 1]
```diff
@@ -62,7 +62,7 @@ static int diff_file_stats_full_to_buf(
 	old_size = delta->old_file.size;
 	new_size = delta->new_file.size;
 
-	if (strcmp(old_path, new_path) != 0) {
+	if (old_path && strcmp(old_path, new_path) != 0) {
```
[the remark this answers is not in the document]
Comment (ddevault) [reply src/diff_stats.c:65]: Good catch, I'll push a revised patch.

**Comment by ethomson** [comment]
Thanks @ddevault - sorry about the delay 😢

**Review by ethomson — APPROVED** [review]

**Comment by ddevault** [comment]
Thanks!

**Pull request merged**